### PR TITLE
Fix for https://github.com/saltstack/salt/issues/38472

### PR DESCRIPTION
**Pull request by jinm**
### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Tests written?

Yes/No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

**Comment by rallytime** [comment]
Refs #38472

File: salt/modules/file.py [hunk 1]
```diff
@@ -3818,7 +3818,7 @@ def _get_local_file_source_sum(path):
                 # but `cached_sum == source_sum['hsum']` is elliptical as prev if
                 sfn = cached_dest
                 source_sum = {'hsum': cached_sum, 'hash_type': htype}
-            elif cached_sum != source_sum['hsum']:
+            elif cached_sum != source_sum.get('hsum', __opts__.get('hash_type', 'md5')):
```
Comment (terminalmage) [inline salt/modules/file.py]: We should not be using ``__opts__.get()`` here, nor falling back to md5, for reasons I explain further in [this pull request](https://github.com/saltstack/salt/pull/40552), which I have just opened to fix similar references in the 2016.3 branch.

Please change this to ``__opts__['hash_type']``

**Review by terminalmage — CHANGES_REQUESTED** [review]
See my comments for required changes.

**Comment by terminalmage** [comment]
@rallytime once the requested changes have been made and this has been merged, it will need a backport to 2016.11.

**Comment by jinm** [comment]
@rallytime @terminalmage the change is in.

**Comment by terminalmage** [comment]
There was no need to make the other changes in develop, because https://github.com/saltstack/salt/pull/40552 would have taken care of this when merged forward.

There are a lot of failing tests, but it looks like these might have been broken in develop when you branched. Can you please fetch from our develop branch and rebase?

```
git remote add upstream https://github.com/salstack/salt
git fetch upstream
git rebase upstream/develop
git push origin develop --force
```

**Comment by terminalmage** [comment]
https://github.com/saltstack/salt/pull/40559 seems to have been opened instead of just rebasing this one. Closing in favor of https://github.com/saltstack/salt/pull/40559.

**Pull request closed without merging**